cmd/admission-controller/mutatingwebhook.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/scylladb/go-log"
+	"github.com/scylladb/scylla-operator-autoscaler/pkg/api/v1alpha1"
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// recommendationApplier overwrites ScyllaCluster spec with recomendations given by Recommender (if available)
+type recommendationApplier struct {
+	Client  client.Client
+	decoder *admission.Decoder
+
+	c      client.Client
+	logger log.Logger
+}
+
+func getDataCenterRecommendations(sca *v1alpha1.ScyllaClusterAutoscaler) []v1alpha1.DataCenterRecommendations {
+	if sca.Status.Recommendations == nil {
+		return nil
+	}
+	dcRecs := sca.Status.Recommendations.DataCenterRecommendations
+	if len(dcRecs) == 0 {
+		return nil
+	}
+	return dcRecs
```

Comment:
nil slice and zero len slice should be treated the same, you can simply return `sca.Status.Recommendations.DataCenterRecommendations`